src/main/java/de/tum/in/www1/artemis/web/rest/StudentQuestionAnswerResource.java

```diff
@@ -61,19 +66,28 @@ public StudentQuestionAnswerResource(StudentQuestionAnswerRepository studentQues
     /**
      * POST /question-answers : Create a new studentQuestionAnswer.
      *
+     * @param courseId the id of the course the answer belongs to
      * @param studentQuestionAnswer the studentQuestionAnswer to create
      * @return the ResponseEntity with status 201 (Created) and with body the new studentQuestionAnswer, or with status 400 (Bad Request) if the studentQuestionAnswer has already
      *         an ID
      * @throws URISyntaxException if the Location URI syntax is incorrect
      */
-    @PostMapping("/student-question-answers")
+    @PostMapping("courses/{courseId}/student-question-answers")
     @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
-    // TODO: there are no security checks here. The API endpoint should at least include the course id
-    public ResponseEntity<StudentQuestionAnswer> createStudentQuestionAnswer(@RequestBody StudentQuestionAnswer studentQuestionAnswer) throws URISyntaxException {
+    public ResponseEntity<StudentQuestionAnswer> createStudentQuestionAnswer(@PathVariable Long courseId, @RequestBody StudentQuestionAnswer studentQuestionAnswer)
+            throws URISyntaxException {
```

Comment:
there is still no security check here. Please check that the user is at least a student in the course

Reply:
If you touch this file, there are additional TODOs regarding the REST URLs and the security checks. Could you please fix those for all requests in this class and in the class StudentQuestionResource, so that all URLs and security checks are correct and consistent?
Thank you!